### ISSUE TITLE
[enhancement] Add Docker support

### ISSUE THREAD

**Issue by saulshanabrook** [DamnWidget/anaconda#266]
First, thanks for this amazing project!

I find these days that I always am running my Python code in a [Docker](https://www.docker.com/) container (usually with [Fig](http://www.fig.sh/) as well). 

Would it be possible for anaconda to use the interpreter within that container somehow?

**Comment by saulshanabrook** [comment]
Maybe [use docker exec](http://stackoverflow.com/a/27357289)? 
```
docker exec container_name /usr/bin/python

```

**Comment by DamnWidget** [comment]
Hi.
Honestly, I have no clue about how to use docker with anaconda as I had little use for docker myself. 

In Vagrant we use a special vagrant branch that starts a minimal anaconda's JsonServer and exposes it due port Nat between the host and the guests but I am not sure about this approach with docket I should make some tests and become familiar with docker to be able to give a correct answer to this question.

**Comment by saulshanabrook** [comment]
I was trying to look and see what anaconda needs out of a python
interpreter. For example, I can run "fig run client python" to get an
interpreter.... Would that work? Sorry I don't know much about the
internals of how it uses the interpreter.

On Friday, February 6, 2015, Oscar Campos <notifications@github.com> wrote:

> Hi.
> Honestly, I have no clue about how to use docker with anaconda as I had
> little use for docker myself.
>
> In Vagrant we use a special vagrant branch that starts a minimal
> anaconda's JsonServer and exposes it due port Nat between the host and the
> guests but I am not sure about this approach with docket I should make some
> tests and become familiar with docker to be able to give a correct answer
> to this question.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DamnWidget/anaconda/issues/266#issuecomment-73183507>.
>


-- 

Saul Shanabrook
about.me/shanabrook
[image: Saul Shanabrook on about.me]
  <http://about.me/shanabrook>

**Comment by DamnWidget** [comment]
Hi, take a look here: http://damnwidget.gitbooks.io/anacondast3-developers-documentation/content/architecture/jsonserver_related_design_and_architecture.html

**Comment by DamnWidget** [comment]
And here: http://damnwidget.gitbooks.io/anacondast3-developers-documentation/content/architecture/README.html

**Comment by saulshanabrook** [comment]
I figured out how to hack together some docker support (at least for in file linting).

The process is basically to run the vagrant server from the same docker image you are running your code from, and then point anaconda to that port.

I made [a bash script]() that will clone and run the anaconda vagrant server. To use this with your Docker host, you can execute this script within whatever image you are using to run your code.

```bash
# default python 3 image
$ docker run --rm -it -p 19360:19360 --entrypoint bash python -c 'source <(curl -s https://gist.githubusercontent.com/saulshanabrook/74d3447d2758462e2bf8/raw/977d88f5451d49b73abde8543e235ba0c9c44849/up.py)'

# or if you are building your own image
$ docker build -t my-thing .
$ docker run --rm -it -p 19360:19360 --entrypoint bash  my-thing -c 'source <(curl -s https://gist.githubusercontent.com/saulshanabrook/74d3447d2758462e2bf8/raw/977d88f5451d49b73abde8543e235ba0c9c44849/up.py)'
```

Then create a sublime text project [and configure it to support vagrant](http://damnwidget.github.io/anaconda/vagrant/#toc_5). For the `network`, using the IP of your docker instance (`boot2docker ip` if on a mac`) and the `mode` of `"private"`. For example:

```
{
    "build_systems":
    [
        {
            "name": "Anaconda Python Builder",
            "selector": "source.python",
            "shell_cmd": "python -u \"$file\""
        }
    ],
    "folders":
    [
        {
            "follow_symlinks": true,
            "path": "."
        }
    ],
    "settings":
    {
        "vagrant_environment": {
            "directory": "/Users/saul/",       // Directory where the Vagrantfile is located in your local machine
            "network": {
                "mode": "private",          // configured vagrant network mode
                "address": "192.168.59.103",  // guest machine IP address
                "port": 19360               // guest machine anaconda vagrant_server port
            }
        }
    }
}
```

the `directory` is irrelevant, because their is no vagrantfile. However, you need vagrant installed (`brew cask install vagrant`) or else Sublime Text will crash when trying to lint a python file with these settings.


Now you can get syntax completion/errors for your Docker python environment!

Obviously, it is still very messy, and just hacked together.

**Comment by saulshanabrook** [comment]
@DamnWidget this seems to sorta work, but whenever I type a character, the server logs an error

```
E302
E302
E302
error: uncaptured python exception, closing channel <__main__.JSONHandler connected 192.168.59.3:60400 at 0x7f40f1212dd8> (<class 'TypeError'>:__init__() missing 1 required positional argument: 'html' [/usr/local/lib/python3.4/asyncore.py|read|83] [/usr/local/lib/python3.4/asyncore.py|handle_read_event|442] [/usr/local/lib/python3.4/asynchat.py|handle_read|171] [../anaconda_server/jsonserver.py|found_terminator|93] [../anaconda_server/jsonserver.py|handle_command|111] [/anaconda/anaconda_server/handlers/jedi_handler.py|run|30] [/anaconda/anaconda_server/lib/anaconda_handler.py|run|48] [/anaconda/anaconda_server/handlers/jedi_handler.py|doc|98])
error: uncaptured python exception, closing channel <__main__.JSONHandler connected 192.168.59.3:60403 at 0x7f40f2990588> (<class 'TypeError'>:__init__() missing 1 required positional argument: 'html' [/usr/local/lib/python3.4/asyncore.py|read|83] [/usr/local/lib/python3.4/asyncore.py|handle_read_event|442] [/usr/local/lib/python3.4/asynchat.py|handle_read|171] [../anaconda_server/jsonserver.py|found_terminator|93] [../anaconda_server/jsonserver.py|handle_command|111] [/anaconda/anaconda_server/handlers/jedi_handler.py|run|30] [/anaconda/anaconda_server/lib/anaconda_handler.py|run|48] [/anaconda/anaconda_server/handlers/jedi_handler.py|doc|98])
error: uncaptured python exception, closing channel <__main__.JSONHandler connected 192.168.59.3:60406 at 0x7f40f11c86d8> (<class 'TypeError'>:__init__() missing 1 required positional argument: 'html' [/usr/local/lib/python3.4/asyncore.py|read|83] [/usr/local/lib/python3.4/asyncore.py|handle_read_event|442] [/usr/local/lib/python3.4/asynchat.py|handle_read|171] [../anaconda_server/jsonserver.py|found_terminator|93] [../anaconda_server/jsonserver.py|handle_command|111] [/anaconda/anaconda_server/handlers/jedi_handler.py|run|30] [/anaconda/anaconda_server/lib/anaconda_handler.py|run|48] [/anaconda/anaconda_server/handlers/jedi_handler.py|doc|98])
error: uncaptured python exception, closing channel <__main__.JSONHandler connected 192.168.59.3:60408 at 0x7f40f1020d68> (<class 'TypeError'>:__init__() missing 1 required positional argument: 'html' [/usr/local/lib/python3.4/asyncore.py|read|83] [/usr/local/lib/python3.4/asyncore.py|handle_read_event|442] [/usr/local/lib/python3.4/asynchat.py|handle_read|171] [../anaconda_server/jsonserver.py|found_terminator|93] [../anaconda_server/jsonserver.py|handle_command|111] [/anaconda/anaconda_server/handlers/jedi_handler.py|run|30] [/anaconda/anaconda_server/lib/anaconda_handler.py|run|48] [/anaconda/anaconda_server/handlers/jedi_handler.py|doc|98])
error: uncaptured python exception, closing channel <__main__.JSONHandler connected 192.168.59.3:60410 at 0x7f40f1020eb8> (<class 'TypeError'>:__init__() missing 1 required positional argument: 'html' [/usr/local/lib/python3.4/asyncore.py|read|83] [/usr/local/lib/python3.4/asyncore.py|handle_read_event|442] [/usr/local/lib/python3.4/asynchat.py|handle_read|171] [../anaconda_server/jsonserver.py|found_terminator|93] [../anaconda_server/jsonserver.py|handle_command|111] [/anaconda/anaconda_server/handlers/jedi_handler.py|run|30] [/anaconda/anaconda_server/lib/anaconda_handler.py|run|48] [/anaconda/anaconda_server/handlers/jedi_handler.py|doc|98])
error: uncaptured python exception, closing channel <__main__.JSONHandler connected 192.168.59.3:60412 at 0x7f40f1020f98> (<class 'TypeError'>:__init__() missing 1 required positional argument: 'html' [/usr/local/lib/python3.4/asyncore.py|read|83] [/usr/local/lib/python3.4/asyncore.py|handle_read_event|442] [/usr/local/lib/python3.4/asynchat.py|handle_read|171] [../anaconda_server/jsonserver.py|found_terminator|93] [../anaconda_server/jsonserver.py|handle_command|111] [/anaconda/anaconda_server/handlers/jedi_handler.py|run|30] [/anaconda/anaconda_server/lib/anaconda_handler.py|run|48] [/anaconda/anaconda_server/handlers/jedi_handler.py|doc|98])
```

**Comment by DamnWidget** [comment]
@saulshanabrook do you get this error if you try to run it in a vagrant environment?

**Comment by pikeas** [comment]
+1, would really love this feature. I do all my development in Docker containers, but this means I miss out on autocomplete.

**Comment by saulshanabrook** [comment]
@DamnWidget I don't know, sorry I don't use Vagrant.

**Comment by aldanor** [comment]
+1 on the docker, it seems to be quite a popular dev option these days...

**Comment by szuliq** [comment]
I'm also missing this feature. :+1:

**Comment by mrmachine** [comment]
This would be great. I believe PyCharm supports "remote debugger" which can be used with the debugger running inside a Docker container. But I much prefer Sublime Text and Anaconda to PyCharm.

**Comment by astrikov-d** [comment]
+1 for this feature. It would be great to have autocompletion and go-to features based on the python interpreter inside Docker container.

**Comment by DamnWidget** [comment]
I will take a look at this to see if it could be included in v2.0.0 (next major release), as I see that this is a popular request with all the +1's in here. 

Said that, I would probably don't ever use the feature myself ever so we would need a maintainer for it, we already have problems with the vagrant support as no one takes care of the vagrant branch.

I don't like to have dead things in the package.

**Comment by DamnWidget** [comment]
Hi guys.

I did my first tests and I had excellent results, my understanding about docker since the first post in this thread was written has grown a lot, what a fantastic tool by the way, I am pleased to announce that I was able to create a small docker image based in Alpine Linux with  Python 2.7.10 on it running a minimal jsonserver with anaconda v1.4.25 on it exposing it's port and printing the JsonServer log to it's standard output so it's available via `docker logs` command.

What was working? Everything that works in vagrant support that is:
Linting 100%
Completion 100%
McCabe/Code Analyzer: 100%
Goto: 100% on files that existed in the same path in both systems (not so useful, I will probably try to make something smart here to allow gotos and find usages)
DocStrings and Tooltips: 100%
Imports Checker: 0% (I will probably deactivate it completely when running in remote)

Next step from here is a refactor of the Worker class to make it's interface completely generic so we will end with something like DockerWorker and VagrantWorker both inheriting from RemoteWorker that inherits himself from Worker.

That means that in version v2.0.0 the anaconda JsonServer is gonna be able to run from:
Localhost
Vagrant VMs
Docker Containers
Any remote host that you can run it and connect to over the internet

Note that probably the third option is gonna be kinda slow.

A better/smarter failback mechanism must be put in place to avoid freeze the ST3 context if the remote server is not reachable for whatever reason or the configured python interpreter is not available.

I will push a branch with the changes so you can test it and give feedback/contribute as soon as I can.

**Comment by DamnWidget** [comment]
Hi guys, I am still designing the Docker support but I already added vanilla remote worker support and vagrant support. Even if the docker support is not there yet, you can make an idea of how is gonna be taking a look at the new wiki pages for anaconda v2

https://github.com/DamnWidget/anaconda/wiki/Anaconda-V2:-Using-a-remote-python-interpreter
https://github.com/DamnWidget/anaconda/wiki/Anaconda-v2:-Using-a-remote-python-interpreter-through-Vagrant

Just take a look and share your thoughts, suggestions as the docker configuration support is gonna be like that.

What do you think?

**Comment by DamnWidget** [comment]
I will push the in-development `docker` branch tomorrow so you can start testing it and giving feedback as soon as possible, I am pretty sure that you will find bugs. I have still to make a huge refactor of the `anaconda_lib` directory including the new written `workers` so another beta test branch will be released after the refactor.

You can find the documentation here in the wiki page [Using a Python interpreter running in a Docker Container](https://github.com/DamnWidget/anaconda/wiki/Anaconda-v2:-Using-a-remote-python-interpreter-through-Docker)

**Comment by DamnWidget** [comment]
New branch is pushed: https://github.com/DamnWidget/anaconda/tree/docker 

To install just clone the repo into your Sublime Text 3 Packages directory, (don't forget to uninstall the Package Control anaconda installation first) then do a checkout of the `docker` branch.

To report issues in the docker branch open a new issue with title [Docker Support] <Whatever> and add a new comment in this issue pointing to the new issue that you created.

Thanks.

**Comment by DamnWidget** [comment]
Hi guys.

I pushed new commits with a complete refactor of every worker (Local, Remote and Vagrant) the way that the workers are used has not changed at all so the previous documentation is still valid. I need you guys test it again.

The refactor makes the code more maintainable and reduces the ST3 gui freezes when there are problems to connect the plugin to the JsonServer/Minserver.

Any feedback is welcome (I didn't got any feedback so far that's not good).

Please, do a git pull in your `docker` branch.

Thanks.

**Comment by mrmachine** [comment]
@damnwidget Thanks very much for the flurry of activity on getting this feature working. I've been following your posts with great anticipation and hope to test this in the next few days.

**Comment by astrikov-d** [comment]
@DamnWidget Thank you!

**Comment by mrmachine** [comment]
I switched to the docker branch, and now after starting Sublime Text I get:

> <Anaconda.anaconda_lib.workers.local_worker.LocalWorker object at 0x10821a450> initial check failed because:

> can not connect to localhost in port 49503

> check that there is Python process executing the anaconda jsonserver.py script running in your system. If there is, check that you can connect to your localhost writing the following script in your Sublime Text 3 console:

> import socket;socket.socket(socket.AF_INET, socket.SOCK_STTREAM).connect(("localhost", 49503))

> Note: the port is the first number of the Python process that you found earlier

**Comment by DamnWidget** [comment]
Did you restarted your ST3 after switch branch?
Did you follow the instructions on the error message?

**Comment by mrmachine** [comment]
I did just now (but the error happened a while ago -- I tried to restart ST3 but the error didn't happen again). I don't see a Python process executing `jsonserver.py` on my system.

**Comment by mrmachine** [comment]
I ran this inside an existing running container (with anaconda bind mounted into the container and port 19360 exposed on the host):

```
$ /opt/anaconda/anaconda_server/docker/start python 19360 docker_project /opt/sfmoma
tail: cannot open '/opt/sfmoma/var/.local/share/anaconda/logs/docker_project/anaconda_jsonserver.log' for reading: No such file or directory
tail: no files remaining
```

When I ran it again, it said it was already running, and the log file does not exist. So I guess that's just a startup issue.

Then I configured my Python interpreter as:

```
tcp://172.16.144.137:19360?pathmap=~/IC/Projects/sfmoma,/opt/sfmoma
```

But when I try to use a feature like Goto object definition (CTRL-OPT-G), nothing happens.

I can telnet to port 19360 on my docker host to check the connection, and when I sent some random characters to the connection I got an error inside the container.

On OS X host:

```
telnet $(dinghy ip) 19360                                                                                                                                staging ✱ ◼
Trying 172.16.144.137...
Connected to 172.16.144.137.
Escape character is '^]'.

fd
Connection closed by foreign host.
```

Inside container:

```
error: uncaptured python exception, closing channel <__main__.JSONHandler connected 172.16.144.1:62263 at 0x7f60566afbd8> (<type 'exceptions.TypeError'>:string indices must be integers, not str [/usr/lib/python2.7/asyncore.py|read|83] [/usr/lib/python2.7/asyncore.py|handle_read_event|449] [/usr/lib/python2.7/asynchat.py|handle_read|165] [/opt/anaconda/anaconda_server/minserver.py|found_terminator|79])
```

**Comment by DamnWidget** [comment]
There is some points that need clarification here:

1) Make sure that your branch is up to date and this one is your last [docker branch commit](https://github.com/DamnWidget/anaconda/commit/06210bf65de7beb0c456a313e0585d65efa2769e)
2) The $HOME environment variable is pointing to `/opt/sfmoma/var` make sure that the user that executes `python` to start the anaconda's `minserver` has the right permissions to write to that directory
3) If you are exposing the port `19360` why you are using the IP `172.16.144.137` to connect to? Should not it be `localhost`?
4) The error is because you just send random garbage from telnet and the `minserver` speaks Anaconda's JSON protocol

**Comment by mrmachine** [comment]
1) That is exactly the commit I am using.
```
```$ git describe
v1.4.26-14-g06210bf
```

2) $HOME is pointing to /opt/sfmoma/var and the same use who owns that directory ran anacond'a `minserver`. The log file was written to -- just on the first startup, when the file didn't exist, I got that error, but it did start and is writing to the log.

3) I'm on OS X, so the port that is being exposed gets exposed on the "docker machine" VM (the IP 172.16.144.137), not localhost. Since Docker Engine runs inside the VM on OS X, it can't expose any ports on `localhost`.

4) Yep. I was just verifying that the connection from OS X to `minserver` was working.

**Comment by DamnWidget** [comment]
Ooops my fault, I didn't realized that you were working under OS X but you commented it in your previous post.

I will take a look this evening when I can have access to my OS X box in home but it's weird, it should work with no problem at all, I even used a remote worker running in my work Linux box from my Windows home box over the Internet with no problems, the `tcp://` Worker works like a charm normally.

**Comment by mrmachine** [comment]
Here's the log:

```
2016-06-01 07:39:49,835: DEBUG   : bind: address=('0.0.0.0', 19360)
2016-06-01 07:39:49,835: DEBUG   : listen: backlog=5
2016-06-01 07:39:49,843: INFO    : Anaconda Server started in port 19360 with cache dir /opt/sfmoma/var/.cache/jedi/docker_project and extra paths /opt/sfmoma
2016-06-01 07:41:07,435: ERROR   : Traceback (most recent call last):
  File "/opt/anaconda/anaconda_server/minserver.py", line 236, in <module>
    server = JSONServer(('0.0.0.0', port))
  File "/opt/anaconda/anaconda_server/minserver.py", line 132, in __init__
    self.bind(self.address)
  File "/usr/lib/python2.7/asyncore.py", line 342, in bind
    return self.socket.bind(addr)
  File "/usr/lib/python2.7/socket.py", line 224, in meth
    return getattr(self._sock,name)(*args)
error: [Errno 98] Address already in use
2016-06-01 07:41:07,436: ERROR   : [Errno 98] Address already in use
2016-06-01 07:47:31,363: INFO    : Incomming connection from ('172.16.144.1', 62263)
2016-06-01 07:47:33,168: INFO    : No data received in the handler
2016-06-01 08:02:14,307: ERROR   : Traceback (most recent call last):
  File "/opt/anaconda/anaconda_server/minserver.py", line 236, in <module>
    server = JSONServer(('0.0.0.0', port))
  File "/opt/anaconda/anaconda_server/minserver.py", line 132, in __init__
    self.bind(self.address)
  File "/usr/lib/python2.7/asyncore.py", line 342, in bind
    return self.socket.bind(addr)
  File "/usr/lib/python2.7/socket.py", line 224, in meth
    return getattr(self._sock,name)(*args)
error: [Errno 98] Address already in use
2016-06-01 08:02:14,307: ERROR   : [Errno 98] Address already in use
2016-06-01 08:11:07,633: DEBUG   : bind: address=('0.0.0.0', 19360)
2016-06-01 08:11:07,633: DEBUG   : listen: backlog=5
2016-06-01 08:11:07,636: INFO    : Anaconda Server started in port 19360 with cache dir /opt/sfmoma/var/.cache/jedi/docker_project and extra paths /opt/sfmoma
```

Is there a command I can paste into the ST3 console to verify the connection is established and that it's actually working, besides just trying Go to object definition?

**Comment by DamnWidget** [comment]
Ok, answering your first question, you can use "Anaconda: Show Documentation" or you can just write whatever invalid random string that makes a syntax error and see if it gets linted, you can also just try to trigger the autocompletion in a standard lib module like `os`, if any of that is success, then you are on the road and ready to rock.

About the paths. Anaconda always had the concept of `extra_path`, the `extra_path` is used to tell the underlying `jedi` library that we want to look for code in non standard places. The remote workers (`tcp://`, `vagrant://`) introduce a new paths concept, the `pathmap`, they are very different between them, the `pathmap` just translates your local paths into remote paths. Let's say that your code resides in `/Users/mrmachine/projects/myproject` but in the docker container it resides in `/opt/myproject`, when the ST3 client sends a command that uses paths, it sends the local path so for example `/Users/mrmachine/projects/myproject/lib/somethingcool.py` that path does not exists in the docker container, the `pathmap` translates this into `/opt/myproject/lib/somethingcool.py`

You still need to pass the `/opt/myproject` path to the Jedi, to do so you can add the query option `&extra=path1&extra=path2&extra=pathN...` as part of your `tcp://` URI and add as many extra paths as you need.

**Comment by DamnWidget** [comment]
@mrmachine, I just tested in my OS X and everything was working fine, I am using El Capitan and the new Docker Beta for OS X (that is not Docket Toolbox) it's much cleaner.

I can't reproduce your error, anyway, something that I noticed recently in your report is the error message in your ST3 console is for LocalWorkers exclusively, if you are using a RemoteWorker trough the `tcp://` URI you should get an error telling you to check your internet connection and other things related with the remote host.

Are you sure there is no per-project or `.rcfile` messing around with your options?

**Comment by DamnWidget** [comment]
Hey guys, how are we looking? Is anyone giving a try to the v2.0.0-rc1 or docker branches? It looks fine, any concerns? Release of v2.0.0 is closer now so all the feedback would be much appreciated.

**Comment by mrmachine** [comment]
I will try again. I gave up for a while after the last issue.

**Comment by ik-dev** [comment]
I was configuring my anaconda python interpreter on windows after starting minserver on one of the containers. But anaconda is failing to find the connection.

Logs on ST3:
> ERROR:root:Your operating system denied the spawn of tcp://192.168.99.100:9999 process. Make sure your configured interpreter is a valid python binary executable and is in the PATH
> The OS did return [WinError 2] The system cannot find the file specified

Minserver logs:

> 2016-07-15 12:58:34,085: DEBUG   : bind: address=('0.0.0.0', 9999)
> 2016-07-15 12:58:34,085: DEBUG   : listen: backlog=5
> 2016-07-15 12:58:34,086: INFO    : Anaconda Server started in port 9999 with cache dir /root/.cache/jedi
> 

Here is my anaconda user settings:
```
{
	"python_interpreter": "tcp://192.168.99.100:9999"
}
```

Can anyone let me know what's missing?

**Comment by DamnWidget** [comment]
Probably ST3 already fired a local anaconda Worker instead of a RemoteWorker and it was stick to it.

If you can reproduce it, open a ST3 console and write the following:
```python
from anaconda.anaconda_lib.workers.market import Market
print(Market)
```
In that way we can see the kind of worker that your ST3 launched for you. In any case, I see that you are using the `docker` branch, try to use the `v2.0.0-rc1` branch as it has some bug fixes already.

I remember that I did fix some issues related with the interpreter/project switcher in `v2.0.0-rc1`

**Comment by ik-dev** [comment]
Tried `v2.0.0-rc1` branch but `minserver` failed to start with following stack trace:

```
 Traceback (most recent call last):
   File "/opt/anaconda/anaconda_server/minserver.py", line 28, in <module>
     from lib.log import log_directory
 ImportError: No module named log
bot_anaconda_1 exited with code 1
```

On ST3 I tried to import anaconda module and even that failed.

```
ERROR:root:Your operating system denied the spawn of tcp://192.168.99.100:9999 process. Make sure your configured interpreter is a valid python binary executable and is in the PATH
The OS did return [WinError 2] The system cannot find the file specified
from anaconda.anaconda_lib.workers.market import Market
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named 'anaconda'
```

PS: It would be great if you could share  `Dockerfile` and `docker-compose.yml` meant only for anaconda server.

**Comment by DamnWidget** [comment]
Looks like your ST3 is not using the Git version of anaconda at all, make sure that you uninstall any Anaconda version downloaded by Package Control, remove the git repository from the `Packages` directory and clone it again.

I will add a Gist with the config files to the wiki

**Comment by ik-dev** [comment]
I had installed anaconda package which was available via Package Control. Removed it, cloned `v2.0.0-rc1` and restarted ST3. Now it seems to be working.

```
>>> from anaconda.anaconda_lib.workers.market import Market
>>> print(Market)
1 workers in the market

Window ID: 2
<anaconda.anaconda_lib.workers.remote_worker.RemoteWorker object at 0x0000002F873EB588>
```

Let me know if I can help by contributing to that wiki or somehow.

Thanks for your help @DamnWidget :+1:

**Comment by DamnWidget** [comment]
Here the gist with the example and instructions:
https://gist.github.com/DamnWidget/3e58cec118879ed7baa08dc283e162a0

**Comment by DamnWidget** [comment]
How are we looking guys? Are you still testing this? Shall we release and go 🌴? :P

**Comment by villlem** [comment]
Looks good to me. Using it every day.

út 16. 8. 2016 v 14:40 odesílatel Oscar Campos <notifications@github.com>
napsal:

**Comment by DamnWidget** [comment]
Docker support has been released with the anaconda v2.0.0 release today
